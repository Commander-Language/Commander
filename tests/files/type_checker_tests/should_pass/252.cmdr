//Test that lambdas type check
add = fn () -> {
    println 3 + 4;
};
add();
x = add();
