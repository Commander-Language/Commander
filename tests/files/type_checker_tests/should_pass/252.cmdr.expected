(PrgmNode
	(ExprStmtNode (BinOpExprNode (BindingNode add) SET (LambdaExprNode (BindingsNode) (ScopeStmtNode (StmtsNode
	(PrintlnStmtNode (BinOpExprNode (IntExprNode 3 INT) ADD (IntExprNode 4 INT) INT)))) (fn () -> VOID)) (fn () -> VOID)))
	(ExprStmtNode (CallExprNode (LValueExprNode (VarExprNode add (fn () -> VOID)) (fn () -> VOID)) (ExprsNode) VOID))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (CallExprNode (LValueExprNode (VarExprNode add (fn () -> VOID)) (fn () -> VOID)) (ExprsNode) VOID) VOID)))
